feat: add support for ignoring existing patch with `--ignore-existing`

HiDeoo opened the pull request:
First, thanks for the tool, it makes the `pnpm patch` workflow much more pleasant.

### Description

In `pnpm@7.25.0`, [a change](https://github.com/pnpm/pnpm/pull/5906) was made to `pnpm patch` to apply an existing patch when patching a dependency that is already patched *(note: this only works when using the `pnpm patch <pkg name>@<version>` syntax, not the `pnpm patch <pkg name>` syntax)*.

This change works great with `pnpm-patch-i` altho, it also introduced a new option `--ignore-existing` to explicitely disable that new behavior and ignore an existing patch.

This pull request adds support for this option to `pnpm-patch-i`. I used the same option name as `pnpm patch` to keep things simple.

### Additional context

Here is a video demo of running `pnpx pnpm-patch-i is-odd@3.0.1` on `is-odd` with an already existing patch:

<video src="https://user-images.githubusercontent.com/494699/213152476-4e0a501d-61ae-40b1-91c4-3cf3dd2a769e.mp4"></video>

Here is a video demo of running `pnpx pnpm-patch-i --ignore-existing is-odd@3.0.1` on `is-odd` with an already existing patch:

<video src="https://user-images.githubusercontent.com/494699/213153212-ad48460c-aca7-428e-ad0f-2199c045d1e9.mp4"></video>

antfu commented on src/cli.ts:
```diff
@@ -1,8 +1,12 @@
-  console.error('$ pnpm-patch-i <package-name>')
+const args = process.argv.slice(2)
+const name = args[args.length === 1 ? 0 : 1]
+const ignoreExisting = args[0] === '--ignore-existing'
```
I think maybe we should forward all the arguments to it.

HiDeoo replied:
I hesitated between the 2 approaches, I'll update the PR.

HiDeoo commented:
I updated the PR to forward extra arguments to `pnpm patch` so that we can use `--ignore-existing`. The `--edit-dir` option is always overridden by the one generated by `pnpm-patch-i`.

I also removed the README changes as it feels irrelevant with this approach.
